components/service/experiments/src/main/java/mozilla/components/service/experiments/ExperimentsUpdater.kt

```diff
@@ -40,44 +23,22 @@ internal class ExperimentsUpdater(
     private val logger: Logger = Logger(LOG_TAG)
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal var source: ExperimentSource? = null
+    internal var source: KintoExperimentSource? = null
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    @Volatile internal var updateJob: Job? = null
 
     fun initialize(configuration: Configuration) {
-        source = getSource(configuration)
+        source = getExperimentSource(configuration)
 
-        GlobalScope.launch(IO) {
+        updateJob = GlobalScope.launch(IO) {
             // Update the experiments list from the server async, without blocking
             // the app launch.
             updateExperiments()
-
-            // Schedule future updates of the experiments list every few hours.
-            schedule()
         }
     }
 
-    /**
-     * Schedule updating the experiments regularly.
-     *
-     * @param worker worker class
-     */
-    fun schedule() {
-        /*
-        val constraints = Constraints.Builder()
-            .setRequiredNetworkType(NetworkType.CONNECTED)
-            .build()
-        */
-        /*
-        val syncWork = PeriodicWorkRequest.Builder(SyncWorker::class.java, UPDATE_INTERVAL_HOURS, TimeUnit.HOURS)
-            .addTag(TAG)
-            .setConstraints(constraints)
-            .build()
-        WorkManager.getInstance().enqueueUniquePeriodicWork(TAG, ExistingPeriodicWorkPolicy.KEEP, syncWork)
-        */
-        //WorkManager.getInstance().enqueue(OneTimeWorkRequest.from(SyncWorker::class.java))
-    }
-
-    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
```

Comment:
nit, unless you are planning on using this outside of this class, you could leave the `@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)` annotation.